detection_rules/config.py

```diff
@@ -28,7 +28,8 @@ class UnitTest:
     test_only: Optional[List[str]] = None
 
     def __post_init__(self):
-        assert not (self.bypass and self.test_only), 'Cannot use both test_only and bypass'
+        assert not (self.bypass is not None and self.test_only is not None), \
+            'Cannot set both `test_only` and `bypass` in test_config!'
```

Comment:
small nitpick: the double negatives are trickier to understand, so 
```python
assert (self.bypass is None or self.test_only is None)
```
might be better